File: ch02_neutronDiffusion/matlab/adjacency_graph.m
clear
close all

LW = 2;
FS = 20;
FN = 'Times New Roman';

node = [...
    0.0,0.0;   % 5
    1.0,1.0;   % 96
    -1.0,1.0;  % 8
    -0.3,-0.1; % 200
    0.5,-0.3]; % 48

element = [...
    1,2,3;
    1,3,4;
    1,4,5;
    1,5,2];

figure
hold on
trimesh(element,node(:,1),node(:,2),'-o','Color','k','MarkerFaceColor','k','LineWidth',LW)
text(node(1,1),node(1,2)+0.1,'5','FontSize',FS,'FontName',FN)
text(node(2,1),node(2,2)+0.1,'96','FontSize',FS,'FontName',FN)
text(node(3,1),node(3,2)+0.1,'8','FontSize',FS,'FontName',FN)
text(node(4,1),node(4,2)-0.1,'200','FontSize',FS,'FontName',FN)
text(node(5,1),node(5,2)-0.1,'48','FontSize',FS,'FontName',FN)
hold off
set(gca,'Visible','off')
print('../figs/adjacency_graph.eps','-depsc2')
close(gcf)

File: ch02_neutronDiffusion/python/adjacency_graph.py
import numpy as np
import matplotlib.pyplot as plt

LW = 2
FN = 'Times New Roman'
FS = 20
plt.rc('lines', lw=LW)
plt.rc('mathtext', fontset='stix')
plt.rc('font', family=FN, size=FS)

node = np.array([
    [0.0,0.0],   # 5
    [1.0,1.0],   # 96
    [-1.0,1.0],  # 8
    [-0.3,-0.1], # 200
    [0.5,-0.3]]) # 48

element = np.array([
    [1,2,3],
    [1,3,4],
    [1,4,5],
    [1,5,2]])
element -= 1 # adjust for zero-based index

plt.figure()
plt.triplot(node[:,0], node[:,1], element, '-ko')
plt.text(node[0,0], node[0,1]+0.1, '5')
plt.text(node[1,0], node[1,1]+0.1, '96')
plt.text(node[2,0], node[2,1]+0.1, '8')
plt.text(node[3,0], node[3,1]-0.15, '200')
plt.text(node[4,0], node[4,1]-0.1, '48')
plt.xlim([-1.1,1.1])
plt.axis('off')
plt.tight_layout()
plt.savefig('../figs/adjacency_graph.pdf', bbox_inches='tight', pad_inches=0)
plt.show()
